### ISSUE TITLE
Mono Compatibility Broken with AMQP Net Lite Upgrade

### ISSUE THREAD

**Issue by Appareo** [ppatierno/azuresblite#7]
It appears that the upgrade to AMQP NET Lite broken Mono compatibility; AMQP NET Lite no longer lists Mono compatibility as they are moving towards DNX/Core. Pulling source and downgrading to AMQP Net Lite 1.0.0 solved the compatibility issue.

An example of the stack trace I was seeing:
mono-functional_1   | System.Net.Sockets.SocketException : Invalid arguments
mono-functional_1   |   at System.Net.Sockets.Socket.BeginMConnect (System.Net.Sockets.SocketAsyncResult sockares) <0x41186900 + 0x001df> in <filename unknown>:0 
mono-functional_1   |   at System.Net.Sockets.Socket.BeginConnect (System.Net.IPAddress[] addresses, Int32 port, System.AsyncCallback callback, System.Object state) <0x41186200 + 0x00107> in <filename unknown>:0 
mono-functional_1   |   at System.Net.Sockets.Socket.BeginConnect (System.String host, Int32 port, System.AsyncCallback callback, System.Object state) <0x41183ef0 + 0x0008f> in <filename unknown>:0 
mono-functional_1   |   at Amqp.TcpTransport+<>c__DisplayClass3.<ConnectAsync>b__0 (System.AsyncCallback c, System.Object s) <0x41183e40 + 0x00073> in <filename unknown>:0 
mono-functional_1   |   at System.Threading.Tasks.TaskFactory`1[TResult].FromAsyncImpl (System.Func`3 beginMethod, System.Func`2 endFunction, System.Action`1 endAction, System.Object state, TaskCreationOptions creationOptions) <0x411833e0 + 0x002e8> in <filename unknown>:0 
mono-functional_1   | --- End of stack trace from previous location where exception was thrown ---
mono-functional_1   |   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () <0x4118bb30 + 0x00033> in <filename unknown>:0 
mono-functional_1   |   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) <0x4118b560 + 0x0010b> in <filename unknown>:0 
mono-functional_1   |   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) <0x4118b3a0 + 0x000b3> in <filename unknown>:0 
mono-functional_1   |   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) <0x4118b340 + 0x00042> in <filename unknown>:0 
mono-functional_1   |   at System.Runtime.CompilerServices.TaskAwaiter.GetResult () <0x4118b310 + 0x0001a> in <filename unknown>:0 
mono-functional_1   |   at Amqp.TcpTransport.Connect (Amqp.Connection connection, Amqp.Address address, Boolean noVerification) <0x41145a60 + 0x0015e> in <filename unknown>:0 
mono-functional_1   |   at Amqp.Connection.Connect (Amqp.Sasl.SaslProfile saslProfile, Amqp.Framing.Open open) <0x41145730 + 0x0006b> in <filename unknown>:0 
mono-functional_1   |   at Amqp.Connection..ctor (Amqp.Address address, Amqp.Sasl.SaslProfile saslProfile, Amqp.Framing.Open open, Amqp.OnOpened onOpened) <0x4113fa20 + 0x00227> in <filename unknown>:0 
mono-functional_1   |   at Amqp.Connection..ctor (Amqp.Address address) <0x4113f990 + 0x0001b> in <filename unknown>:0 
mono-functional_1   |   at ppatierno.AzureSBLite.Messaging.Amqp.AmqpMessagingFactory.Open (System.String entity) <0x4113f740 + 0x001b7> in <filename unknown>:0 
mono-functional_1   |   at ppatierno.AzureSBLite.Messaging.Amqp.AmqpMessageReceiver.ReceiveEventData () <0x4118d9a0 + 0x00042> in <filename unknown>:0 
mono-functional_1   |   at ppatierno.AzureSBLite.Messaging.EventHubReceiver.Receive () <0x4118d530 + 0x0001f> in <filename unknown>:0

**Comment by BenjaminNiemann** [comment]
Yeah, absoulutly right. after i downgraded the AMPQ Lib to 1.1.0.0 it works very well. ;)
